### PR TITLE
Documentation Update: Added instructions for using the library in App Engine

### PR DESCRIPTION
**Pull request by hiranya911**
Related to #169

### CODE REVIEW

File: docs/user-guide.rst [hunk 1]
```diff
@@ -127,10 +127,33 @@ credentials one of two ways:
 
         credentials = app_engine.Credentials()
 
+In order to make authenticated requests in the App Engine environment using the
+credentials and transports provided by this library, you need to follow a few
+additional steps:
+
+1. If you are using the :mod:`google.auth.transport.requests` transport, vendor
```
Comment (dhermes) [inline docs/user-guide.rst]: Prefer `#.` in RST?

File: docs/user-guide.rst [hunk 2]
```diff
@@ -127,10 +127,33 @@ credentials one of two ways:
...
+   in the ``requests-toolbelt`` library into you app, and enable the App Engine
```
Comment (dhermes) [inline docs/user-guide.rst]: Add link for `requests-toolbelt`?

File: docs/user-guide.rst (hunk 1 above)
Comment (hiranya911) [inline docs/user-guide.rst]: Done

File: docs/user-guide.rst (hunk 2 above)
Comment (hiranya911) [inline docs/user-guide.rst]: Done

**Comment by jonparrott** [comment]
Thanks, @hiranya911 and @dhermes!

**Pull request merged**